app/models/pregnancy.rb

```diff
@@ -67,4 +67,35 @@ def recent_calls
   def old_calls
     calls.order('created_at DESC').offset(10)
   end
+
+  def contact_made?
+    calls.each do |call|
+      return true if call.status == "Reached patient"
+    end
+    false
+  end
+
+  # def pledge_status?(status)
+  #   pledges.each do |pledge|
+  #     return true if pledge[status]
+  #   end
+  #   false
+  # end
+
+  def status
+    status =
```

Comment:
I think this is right on -- very readable and straightforward. I think you could even remove the variable setting on L86 if you want to simplify further and just have the strings.